packages/tools/src/utilities/math/polyline/getNormal3.ts

```diff
@@ -0,0 +1,37 @@
+import { vec3 } from 'gl-matrix';
+import { Types } from '@cornerstonejs/core';
+
+/**
+ * Calculate the normal of a 3D planar polyline
+ * @param polyline - Planar polyline in 3D space
+ * @returns Normal of the 3D planar polyline
+ */
+export default function getNormal3(polyline: Types.Point3[]): Types.Point3 {
+  const vecCrossProd = vec3.create();
+  const vecP1P2 = vec3.create();
+  const vecP2P3 = vec3.create();
+  const vecNormal = vec3.create();
+
+  const meanPoint = polyline.reduce(
+    (acc, cur) => vec3.add(acc, acc, cur),
+    vec3.create()
+  );
+
+  vec3.scale(meanPoint, meanPoint, 1 / polyline.length);
+
+  for (let i = 0, len = polyline.length; i < len; i++) {
+    const p1 = polyline[i];
+    // Using ternary instead of % (mod) operator to make it faster
+    const p2Index = i === len - 1 ? 0 : i + 1;
```

Comment:
same comment

Reply:
Refactored to remove all calls to `glMatrix` functions which makes a huge difference (4x faster) when processing thousands of points.